Makes/MakesViewController.m

```diff
@@ -0,0 +1,111 @@
+//
+//  ViewController.m
+//  Makes
+//
+//  Created by Mohamed Ragab on 1/8/17.
+//  Copyright © 2017 Mohamed Ragab. All rights reserved.
+//
+
+#import "MakesViewController.h"
+#import "Make.h"
+#import <AFNetworking/AFNetworking.h>
+#import <Masonry/Masonry.h>
+#import "MakesCollectionViewCell.h"
+#import "MakeDetailViewController.h"
+#import "MakesManager.h"
+#import <SDWebImage/UIImageView+WebCache.h>
+
+
+
+
+@interface MakesViewController ()
+@property (nonatomic, strong) NSArray *makes;
+@property (nonatomic, strong) UICollectionView *makesCollectionView;
+@end
+
+@implementation MakesViewController
+
+- (void)loadMakes {
+    
+    [[MakesManager manager] makesWithsuccess:^(NSArray *makes) {
+        self.makes = makes;
+        [self.makesCollectionView reloadData];
+    } failure:^(NSError *error) {
+        
+    }];
+
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view, typically from a nib.
+    self.makes = [[NSMutableArray alloc]init];
+    self.title = @"Makes";
+    
+    self.view.backgroundColor = [UIColor whiteColor];
+    
+    
+    UICollectionViewFlowLayout *layout=[[UICollectionViewFlowLayout alloc] init];
+    self.makesCollectionView =[[UICollectionView alloc] initWithFrame:self.view.frame collectionViewLayout:layout];
+    [self.makesCollectionView setDataSource:self];
+    [self.makesCollectionView setDelegate:self];
+    UIImageView *backgroundImage = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"tire_pattern"]];
+    self.view.backgroundColor = [UIColor blackColor];
+    self.makesCollectionView.backgroundView = backgroundImage;
+    
+
+    
+    [self.makesCollectionView setBackgroundColor:[UIColor whiteColor]];
+    [self.makesCollectionView registerClass:[MakesCollectionViewCell class] forCellWithReuseIdentifier:@"cellIdentifier"];
+    [self.view addSubview:self.makesCollectionView];
```

Comment:
Please when you have a cell identifier or a constant string which we will use in our code define it as a const sting like this

static NSString * cont MakesCollectionViewCellIdentifer = @"MakesCollectionViewCellIdentifer";